sageresearch-app-sdk/src/test/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepositoryTest.kt

```diff
@@ -0,0 +1,397 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.sageresearch.viewmodel
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.inOrder
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.reset
+import com.nhaarman.mockitokotlin2.spy
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import hu.akarnokd.rxjava.interop.RxJavaInterop.toV1Single
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.sagebionetworks.bridge.android.manager.ActivityManager
+import org.sagebionetworks.bridge.android.manager.ParticipantRecordManager
+import org.sagebionetworks.bridge.rest.model.Message
+import org.sagebionetworks.bridge.rest.model.ScheduledActivity
+import org.sagebionetworks.research.domain.result.interfaces.TaskResult
+import org.sagebionetworks.research.sageresearch.dao.room.ScheduledActivityEntity
+import org.sagebionetworks.research.sageresearch.dao.room.ScheduledActivityEntityDao
+import org.threeten.bp.Instant
+import java.util.Arrays
+import java.util.UUID
+import java.util.concurrent.TimeUnit.SECONDS
+
+class ScheduleRepositoryTest {
+
+    @Rule
+    @JvmField
+    public var testSchedulerRule = TestSchedulerRule()
+
+    @Mock
+    private lateinit var activityManager: ActivityManager
+
+    @Mock
+    private lateinit var participantRecordManager: ParticipantRecordManager
+
+    @Mock
+    private lateinit var scheduledActivityEntityDao: ScheduledActivityEntityDao
+
+    @Mock
+    private lateinit var scheduledRepositorySyncStateDao: ScheduledRepositorySyncStateDao
+
+    private lateinit var scheduleRepository: ScheduleRepository
+
+    @Before
+    fun beforeTest() {
+        MockitoAnnotations.initMocks(this)
+        scheduleRepository = spy(
+                ScheduleRepository(scheduledActivityEntityDao, scheduledRepositorySyncStateDao, activityManager,
+                        participantRecordManager))
+    }
+
+    @Test
+    fun syncSchedules() {
+        // TODO: 2018/10/05 test syncing
+    }
+
+    @Test
+    fun syncFailedSchedules() {
+        // setup
+        val failedSchedules = mock<List<ScheduledActivityEntity>>()
+
+        // DB returns failed schedules
+        `when`(scheduledActivityEntityDao.activitiesThatNeedSyncedToBridge()).thenReturn(failedSchedules)

```

Comment:
What in the world is the apostrophe operator?

Reply:
Haha idk. My kotlin usually is copied straight from samples